spec/acceptance/helpers.rb

```diff
@@ -0,0 +1,75 @@
+# Time constants in seconds
+HOUR =  60 * 60
+DAY  =  24 * HOUR
+YEAR = 365 * DAY
+
+# Execute `date` command on host with optional arguments
+# and get back a Ruby Time object
+#
+# @param [Host]            host   single Beaker::Host to run the command on
+# @param [Array<String>]   args   Array of arguments to be appended to the
+#                                 `date` command
+# @return [Time]    Ruby Time object
+def get_time_on(host, args = [])
+  arg_string = args.join(' ')
+  date = on(host, "date #{arg_string}").stdout.chomp
+  return Time.parse(date)
+end
+
+# Retrieve the CA enddate on a given host as a Ruby time object
+#
+# @param [Host]            host   single Beaker::Host to get CA enddate from
+#
+# @return [Time]    Ruby Time object
+def get_ca_enddate_time_on(host)
+  if host[:roles].include? 'master' then
+    ca_path = '/etc/puppetlabs/puppet/ssl/ca/ca_crt.pem'
+  else
+    ca_path = '/etc/puppetlabs/puppet/ssl/certs/ca.pem'
+  end
+  on(host, "openssl x509 -in #{ca_path} -noout -enddate") do |result|
+    m = result.stdout.match(/notAfter=(.*) (\d{4}) ([A-Z]{3})$/)
+    if m then
+      enddate = "#{m[1]} #{m[3]} #{m[2]}"
+      return Time.parse(enddate)
+    end
+  end
+  return Time.at(0)
```

Comment:
If there's no meaningful return value then we should return nil, rather than returning a synthetic value. If it gets incorrectly used then debugging this will be harder, rather than returning `nil` and having code blow up because we called methods on nil.